pkgs/servers/home-assistant/default.nix

```diff
@@ -0,0 +1,55 @@
+#{ stdenv, buildPythonApplication, fetchPypi, callPackage
+#, requests, pyyaml, pytz, pip, jinja2, voluptuous, typing, aiohttp, yarl, async-timeout, chardet, astral, certifi
+#, sqlalchemy, netdisco, xmltodict, user-agents
+{ stdenv, python3
+, extraPackages ? x: [] }:
+
+let
+  python = python3.override {
+    packageOverrides = self: super: {
+      yarl = super.yarl.overridePythonAttrs (oldAttrs: rec {
+        name = "${oldAttrs.pname}-${version}";
+        version = "0.18.0";
+        src = oldAttrs.src.override {
+          inherit version;
+          sha256 = "11j8symkxh0ngvpddqpj85qmk6p70p20jca3alxc181gk3vx785s";
+        };
+      });
+      aiohttp = super.aiohttp.overridePythonAttrs (oldAttrs: rec {
+        name = "${oldAttrs.pname}-${version}";
+        version = "2.3.7";
+        src = oldAttrs.src.override {
+          inherit version;
+          sha256 = "0fzfpx5ny7559xrxaawnylq20dvrkjiag0ypcd13frwwivrlsagy";
+        };
+      });
+    };
+  };
+  hass-frontend = python.pkgs.callPackage ./frontend.nix { };
```

Comment:
having this inside `packageOverrides` and then `self.callPackage` is I think a bit nicer, but it is not necessary.

Reply:
It also has the advantage of being able to modify the frontend package.